### ISSUE TITLE
Installing Snipe-IT on Windows Server 2012

### ISSUE THREAD

**Issue by JodJC** [snipe/snipe-it#3298]
I am trying to install Snipe-IT on Windows Server 2012. So far I've gotten to step three without any problems. Although I guess I'll only know if my config file needs tweaking once it runs.

However, at step 4, it says "
    Download Composer-Setup.exe from the Composer website.
    Move the Composer-Setup.exe file C:\inetpub\wwwroot\snipe-it
    Open C:\inetpub\wwwroot\snipe-it in Windows Explorer
    Right click and select Composer Install. This will install the dependencies."

I take this to mean, that I should run the composer install in the snipe-it folder, which is fine but I'm not sure if I'm supposed to pull the php.exe file from the snipe-it folder. that .exe file comes with the wampserver.

![image](https://cloud.githubusercontent.com/assets/25664290/22884385/06c5398c-f1cb-11e6-8ebd-0bff6391a545.png)

So I installed it using the php.exe file from the wamp64 folder.

After that I tried to generate the app key but I'm not sure whether to run the command in the MariaDB command prompt or the mysql client. Either way, it doesn't work or I get a fatal error. 

I went ahead and configured IIS but I assume that it won't work until I get steps 4 and 5 sorted out.

Thanks in advance.

**Comment by JodJC** [comment]
![image](https://cloud.githubusercontent.com/assets/25664290/22884690/9f090fd8-f1cc-11e6-87f3-b634cdbf7ad7.png)

This is what I get when I run the command in MariaDB command prompt

**Comment by korio77** [comment]
if you have it running on IIS check in the IIS that you have correctly configured php, you need to add the php manager to the IIS console, this will make your life really easy when you need to deal with php.

**Comment by vjandrea** [comment]
@JodJC Looks like your autoloader has not been generated yet.
Please try to run `composer install` in the same path (C:\inetpub\wwwroot\snipe-it\). Otherwise you may try with `composer update` and `composer dump-autoload -o`.

**Comment by JodJC** [comment]
This is what I got

![image](https://cloud.githubusercontent.com/assets/25664290/22895413/beccac14-f1f2-11e6-89b2-ec6b0367e3ee.png)


When I test the settings, this is what I get

![image](https://cloud.githubusercontent.com/assets/25664290/22895399/b6484a76-f1f2-11e6-98ec-ada405431370.png)

**Comment by JodJC** [comment]
@vjandrea

Thanks for your suggestion. I made some progress but i'm not there yet. The dependencies installed and I generated the app key. However when I try to load the url, I get an error. In the .env file I set the url to asset.example.com, also I kept the host name empty in the site bindings sections in IIS. Any thoughts?

**Comment by vjandrea** [comment]
I never worked with IIS but it looks like you have some file permissions to fix, look at the Details panel in the last screenshot you posted. Then try with another hostname, like assets.dev so you'll avoid messing with an existing domain. Use the same in the site bindings.

**Comment by korio77** [comment]
I have it working and i have the same window in the test connection, all you need to do is right click in the left menu tree and "edit permisions" once there, you need to add the user of the IIS with full control permissions

![image](https://cloud.githubusercontent.com/assets/20576635/22919472/6361fb38-f290-11e6-9a88-67a35a16b7af.png)

If you are getting an error once loading the web its probably because of lavarel or php, can you check the logs of both and paste them here, they will tell you what you are missing.

You can access the php log easily in the php manager window on the IIS server

![image](https://cloud.githubusercontent.com/assets/20576635/22919524/a48b6978-f290-11e6-9e20-1171622b018d.png)


For the lavarel log you can find it on the snipeIT folder inside this route

\storage\logs

**Comment by JodJC** [comment]
@korio77 

I adjusted the permissions, but after reading your post I realized that I didn't have CGI installed, so when I went to PHP manager, so I'm taking care of that now. 

Also, under the general requirements, it lists PHP >= 5.5.9, do you have a recommendation for a php application?

**Comment by JodJC** [comment]
@korio77 @vjandrea 

I think I'm almost there lol. When I launch the site from IIS, It navigates to assets.example.com/setup, which I assume is the pre-flight setup, but it fails to load. I'm not seeing anything recent in the error log under php settings. Enabling a detailed error page doesn't give me anything either.

![image](https://cloud.githubusercontent.com/assets/25664290/22930101/1952592a-f295-11e6-9d8b-f335f2beaffe.png)

Notice the date.........

**Comment by vjandrea** [comment]
Try enabling display_errors in php.ini

**Comment by korio77** [comment]
if you reached the preflight webpage its probably a lavarel error or even an app error.

Check the lavarel log, if you dont see anything there, enable de debug_mode in the .env file and then reload the page, it should tell you where is the problem

**Comment by JodJC** [comment]
@korio77 

I already have debug mode enabled in the .env file. I will send you a private message in the gitter.im/snipe/snipe-it forum.

**Comment by JodJC** [comment]
@korio77 

I already have debug mode enabled in the .env file. Are you open to a remote desktop session to see what I got?

**Comment by korio77** [comment]
Then check the lavarel log, you should have something there.

Usually when i have the problem of a blank screen when loading the web its because the php manager of the IIS have lost the path to the php-cgi.exe file or because wrong permisions on the public folders of the app.

Im buried in work, cant do a remote desktop, sorry

**Comment by JodJC** [comment]
No problem. Thanks for your assistance and patience.

Is the lavarel log located in "C:\inetpub\wwwroot\snipe-it\storage\logs"? There's a txt file but it was last modified on the 9th and there isn't much there. 

In my php manager, I have this,

![image](https://cloud.githubusercontent.com/assets/25664290/22980113/f0f37fb0-f36e-11e6-9736-4a47f9316a9d.png)

I also stopped the default web site. I can send you more print screens if you wish.

**Comment by korio77** [comment]
i think you have an strange config hahahaha.

So you are using at the same time IIS and wamp?¿ check the apache log of the wamp server then to see if there is something there.

If the config is good there should be something in the error logs somewhere.

**Comment by JodJC** [comment]
Desperate times call for desperate measures I guess lol.

I was trying different things. Earlier, I asked which PHP would you recommend. On Snipe-IT's requirements page, they state PHP >= 5.5.9, they provide a link but what would you recommend.

I suspect I may have to delete everything and start over lol

**Comment by korio77** [comment]
sometimes is faster to start over when you know what are you doing than trying to fix something you have used to test stuff.

I currently have V 5.6 working without any problems.

If you want check the iis log and the apache log before you delete everything, right now you need to check who is handling your web request, i dont know what kind of behaviour can create installing at the same time IIS and wamp on the same user session.

**Comment by JodJC** [comment]
I think I'll redo the web server configuration. Can you give me a step-by-step guide on what I should do? Requirements, configurations, etc.? 

The database is fine(for the most part), the config file is ok. We sorted the dependencies, and we generated the app key. So I think it's down to a web server configuration issue. Can you walk me through it? 

Thanks

**Comment by brianjking** [comment]
https://snipe-it.readme.io/docs/installation @JodJC

**Comment by JodJC** [comment]
@brianjking I followed that already. I'm referring to the web server config.

**Comment by korio77** [comment]
Just follow the guide, and tell us if you find any difficulties.

**Comment by JodJC** [comment]
I went through it again, sorted out PHP. For the purposes of getting it to work, I kept the APP_URL as assets.portal.local. Created the site, followed the guide, but it will won't load. The webpage launches to "http://assets.portal.local/setup". When I go to the PHP log, it says, "PHP Warning:  PHP Startup: Unable to load dynamic library 'C:\Program Files\PHP\v7.0\ext\php_mysql.dll' - The specified module could not be found.
 in Unknown on line 0"

I disabled the default website, edited the permissions, used the IP address of the server on port 80, and kept the hostname blank. 

The page loads in firefox, and even though I enable the detailed error pages, I'm not getting any code or details.
